src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -11766,6 +11766,395 @@ public static Vector512<T> Invoke(Vector512<T> t)
         internal readonly struct Log2Operator<T> : IUnaryOperator<T>
             where T : ILogarithmicFunctions<T>
+        {
+            public static bool Vectorizable => (typeof(T) == typeof(double))
+                                            || (typeof(T) == typeof(float));
+
+            public static T Invoke(T x) => T.Log2(x);
+
+            public static Vector128<T> Invoke(Vector128<T> x)
+            {
+#if NET9_0_OR_GREATER
+                if (typeof(T) == typeof(double))
+                {
+                    return Vector128.Log2(x.AsDouble()).As<double, T>();
+                }
+                else
+                {
+                    Debug.Assert(typeof(T) == typeof(float));
+                    return Vector128.Log2(x.AsSingle()).As<float, T>();
+                }
+#else
+                if (typeof(T) == typeof(double))
+                {
+                    return Log2OperatorDouble.Invoke(x.AsDouble()).As<double, T>();
+                }
+                else
+                {
+                    Debug.Assert(typeof(T) == typeof(float));
+                    return Log2OperatorSingle.Invoke(x.AsSingle()).As<float, T>();
+                }
+#endif
+            }
+
+            public static Vector256<T> Invoke(Vector256<T> x)
+            {
+#if NET9_0_OR_GREATER
+                if (typeof(T) == typeof(double))
+                {
+                    return Vector256.Log2(x.AsDouble()).As<double, T>();
+                }
+                else
+                {
+                    Debug.Assert(typeof(T) == typeof(float));
+                    return Vector256.Log2(x.AsSingle()).As<float, T>();
+                }
+#else
+                if (typeof(T) == typeof(double))
+                {
+                    return Log2OperatorDouble.Invoke(x.AsDouble()).As<double, T>();
+                }
+                else
+                {
+                    Debug.Assert(typeof(T) == typeof(float));
+                    return Log2OperatorSingle.Invoke(x.AsSingle()).As<float, T>();
+                }
+#endif
+            }
+
+            public static Vector512<T> Invoke(Vector512<T> x)
+            {
+#if NET9_0_OR_GREATER
+                if (typeof(T) == typeof(double))
+                {
+                    return Vector512.Log2(x.AsDouble()).As<double, T>();
+                }
+                else
+                {
+                    Debug.Assert(typeof(T) == typeof(float));
+                    return Vector512.Log2(x.AsSingle()).As<float, T>();
+                }
+#else
+                if (typeof(T) == typeof(double))
+                {
+                    return Log2OperatorDouble.Invoke(x.AsDouble()).As<double, T>();
+                }
+                else
+                {
+                    Debug.Assert(typeof(T) == typeof(float));
+                    return Log2OperatorSingle.Invoke(x.AsSingle()).As<float, T>();
+                }
+#endif
+            }
+        }
+
+#if !NET9_0_OR_GREATER
```

Comment:
Is the test project set up to run tests against .NET 8, or are we only testing the .NET 9 and netstandard2.0 bits right now? Given the non-trivial code differences involved in all three versions, it'd probably be best if we ran all three sets of tests... I _think_ right now we're only running two, at least locally (maybe somehow CI runs all of them?)

Reply:
I'm unsure about which it runs currently. I agree it'd be beneficial if all are running, even if some are in the outer loop.

Reply:
@ericstj, @ViktorHofer, I understand from Viktor that this is currently not possible. We're going to have a large amount of code in this library that differs between its netstandard, net8, and net9 assets, and really need to run the tests on all three. How can we make that happen in some form?

Reply:
Steve and I had a conversation about this offline. Being able to test previous .NETCoreApp versions is tracked via https://github.com/dotnet/runtime/issues/54639.

> here are couple things that would need to change:
Obviously, all referenced projects / packages need to support that older .NETCoreApp TFM. I.e. RemoteExecutor currently doesn't (for no particular reason). I have a PR open that changes that: [Update Arcade TFMs according to TFM docs by ViktorHofer · Pull Request #14332 · dotnet/arcade (github.com)](https://github.com/dotnet/arcade/pull/14332/files#diff-f474402363c7ba47c119ded0c56143aa9821e404cda6fd9e95da1595f70f116c). TestUtilities.csproj in runtime needs to change as well.
Unfortunately, when testing .NETCoreApp our runtime specific test infra relies on the runtime being produced live. Changes are needed in the RunTests scripts and the msbuild tests files.
If you want to have those tests run in CI as well, changes to sendtohelix projects would be required as they have the same "Current" expectations hardcoded.

> FWIW there are currently two severe issues affecting running tests locally in runtime:
[Local running of libraries tests is broken · Issue #95762 · dotnet/runtime (github.com)](https://github.com/dotnet/runtime/issues/95762)
[`dotnet test` is not running tests for .NETFramework targets · Issue #94183 · dotnet/runtime (github.com)](https://github.com/dotnet/runtime/issues/94183)

Reply:
A workaround might be to have a separate test project that uses the net8.0 build of this assembly src project, but it would be running on latest runtime since that's what our tests do.  The work to run on older frameworks is substantial as @ViktorHofer points out.

Reply:
> A workaround

Thanks, I'll try.

Reply:
https://github.com/dotnet/runtime/pull/96641